python/plugins/processing/gui/wrappers.py

```diff
@@ -0,0 +1,857 @@
+        elif self.dialogType == DIALOG_BATCH:
+            return self.widget.setText(value)
+        else:
+            self.setComboValue(value)
+
+
+    def value(self):
+        if self.dialogType == DIALOG_STANDARD:
+            try:
+                return self.widget.itemData(self.widget.currentIndex())
+            except:
+                return self.widget.getValue()
+        elif self.dialogType == DIALOG_BATCH:
+            return self.widget.getText()
+        else:
+            def validator(v):
+                return bool(v) or self.param.optional
+            return self.comboValue(validator)
+
+class TableFieldWidgetWrapper(WidgetWrapper):
+
+    NOT_SET = '[Not set]'
+
+    def createWidget(self):
+        if self.param.multiple:
+            if self.dialogType == DIALOG_STANDARD:
+                    return MultipleInputPanel(options=[])
+            else:
+                return QLineEdit()
+        else:
+            if self.dialogType == DIALOG_STANDARD:
+                widget = QComboBox()
+                return widget
+            elif self.dialogType == DIALOG_BATCH:
+                widget = QLineEdit()
+                if self.param.default:
+                    widget.setText(self.param.default)
+                return widget
+            else:
+                widget = QComboBox()
+                widget.setEditable(True)
+                fields = self.dialog.getAvailableValuesOfType(ParameterTableField, None)
+                if self.param.optional:
+                    widget.addItem(self.NOT_SET, None)
+                for f in fields:
+                    widget.addItem(self.dialog.resolveValueDescription(f), f)
+                return widget
+
+    def postInitialize(self, wrappers):
+        for wrapper in wrappers:
+            if wrapper.param.name == self.param.parent:
+                layer = wrapper.widget.itemData(wrapper.widget.currentIndex())
+                if layer is not None:
+                    fields = self.getFields(layer, wrapper.param.datatype)
+                    if self.param.multiple:
+                        self.widget.updateForOptions(fields)
+                    else:
+                        self.widget.clear()
+                        if self.param.optional:
+                            self.widget.addItem(self.tr(self.NOT_SET))
+                        self.widget.addItems(fields)
+                break
+
+    def getFields(self, layer, datatype):
+        fieldTypes = []
+        if datatype == ParameterTableField.DATA_TYPE_STRING:
+            fieldTypes = [QVariant.String]
+        elif datatype == ParameterTableField.DATA_TYPE_NUMBER:
+            fieldTypes = [QVariant.Int, QVariant.Double, QVariant.LongLong,
+                          QVariant.UInt, QVariant.ULongLong]
+
+        fieldNames = set()
+        for field in layer.fields():
+            if not fieldTypes or field.type() in fieldTypes:
+                fieldNames.add(unicode(field.name()))
```

Comment:
Ah, that's the reason.
Whatever is there right now will be converted in the PORT_PLUGINS script. This was mainly meant as a heads up because at one point all the code will be switched over and the script will no longer be run.
I'm normally just switching them over as I run into them.